cpp/src/arrow/dataset/partition.h

```diff
@@ -108,6 +108,18 @@ class ARROW_DS_EXPORT PartitionSchemeDiscovery {
   std::shared_ptr<Schema> schema_;
 };
 
+/// \brief Subclass for representing the default, always true scheme.
```

Comment:
Please revert this move. DefaultPartitionScheme is not intended to be public and does not need to be exposed to binding layers